### ISSUE TITLE
LTI 1.3 (Advantage)

### ISSUE THREAD

**Issue by acbart** [pylti/lti#70]
Hello, I saw #31, and I was curious if there was any progress with LTI 1.3 support? So far, I am only aware of [PHP](https://github.com/IMSGlobal/lti-1-3-php-library) and [.NET](https://github.com/andyfmiller/LtiAdvantagePlatform) libraries for the new LTI Advantage standard. Does PyLTI support/plan to support the new standard? Looks like the deprecation schedule has the old style authentication [fading out by December of 2020](https://www.imsglobal.org/lti-security-announcement-and-deprecation-schedule-july-2019).

**Comment by ryanhiebert** [comment]
I haven't done anything with it myself, but I did have a bit of conversation about it at InstructureCon. Would you be interested in helping to get LTI 1.3 into this library? We'd love to have you. We're definitely keen on the idea. Thanks for bringing it up, and letting us know about the timeline!

**Comment by acbart** [comment]
I'm afraid I'm a bit overwhelmed with other projects to be able to contribute any developer time. Hopefully someone else will be able to pitch in!

**Comment by ryanhiebert** [comment]
Fair enough, I'm in a similar situation. My company is keeping an eye on 1.3, but the priority at the moment isn't on implementing it. It sounds like a fun project when we get to it, but I can't promise any timeline. I'll be happy to review any pull requests and help answer questions from anyone that's willing to take on this work.

**Comment by dmitry-viskov** [comment]
@acbart @ryanhiebert hi guys. if you are interested - i've created little library for implementation LTI 1.3/Advantage Tool:
https://pypi.org/project/PyLTI1p3/
https://github.com/dmitry-viskov/pylti1.3 
All suggestions/contribution are welcome

**Comment by ryanhiebert** [comment]
Very interesting. Is that something you'd be interested in talking about merging with this project somehow, or do you think it would be best kept separate?

**Comment by dmitry-viskov** [comment]
@ryanhiebert 

>you'd be interested in talking about merging with this project 

No. I would prefer to keep it separate. My end goal is modification edX (https://github.com/edx/edx-platform) to work as LTI 1.3 Tool. I've tried to find existing LTI 1.3 python solutions but unfortunately didn't find anything except this issue. So I had to re-implement existing PHP library: https://github.com/IMSGlobal/lti-1-3-php-library .

**Comment by ryanhiebert** [comment]
Fair enough. Thanks for telling us about the work you put in! I hope that it works well for your purposes.

**Comment by hari1500** [comment]
> LTI 1.3/Advantage Too

Can we use this to create LTI consumer also? If so, how do we do

**Comment by ryanhiebert** [comment]
I'm not familiar with that project, so I'm afraid I can't help you with that, but you can ask on that GitHub repository: https://github.com/dmitry-viskov/pylti1.3.

There's no LTI 1.3 here yet, but that's just because no-one has implemented it. I'd be happy to review a pull request to that effect.
